### PR TITLE
[dy] AWS codepipeline for deploying Mage

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add terraform templates for deploying Mage to ECS from a CodeCommit repo with AWS CodePipeline. It will create 2 separate CodePipelines, one for building a docker image to ECR from a CodeCommit repository, and another one for reading from ECR and deploying to ECS.

In the `IAM` section, I have an inline policy with certain AWS permissions. I wasn't sure how to narrow the permissions down because I can't figure out exactly what permissions the CodePipeline needs to deploy to ECS. Is there a way to do that without manually trying a bunch of configurations?

# Tests

tested on Mage AWS

cc: @wangxiaoyou1993

### CODE REVIEW

File: aws-code-pipeline/main.tf [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      source  = "hashicorp/aws"
+      version = "~> 4.16"
+    }
+  }
+
+  required_version = ">= 1.2.0"
+}
+
+provider "aws" {
+  region = "us-west-2"
```
Comment (wangxiaoyou1993) [inline aws-code-pipeline/main.tf:13]: use the region from the variable

File: aws-code-pipeline/variables.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  default     = "mage-codepipeline"
+}
+
+variable "ecr_repo_name" {
+  type        = string
+  description = "ECR repo name"
+  default     = "ecr-repo-name"
+}
+
+variable "ecr_image_tag" {
```
Comment (wangxiaoyou1993) [inline aws-code-pipeline/variables.tf:41]: Are we able to use github SHA as image tag so that the docker images are versioned?

**Review by wangxiaoyou1993 — APPROVED** [review]

File: aws-code-pipeline/variables.tf (hunk 2 above)
Comment (dy46) [reply aws-code-pipeline/variables.tf:41]: i can't, the ecr source can only look at one tag

File: aws-code-pipeline/main.tf (hunk 1 above)
Comment (wangxiaoyou1993) [reply aws-code-pipeline/main.tf:13]: @dy46 did you see this comment?

Comment (wangxiaoyou1993) [reply aws-code-pipeline/main.tf:13]: nvm. saw your other PR

**Pull request merged**